### ISSUE TITLE
Creating Interest Grouping is giving error

### ISSUE THREAD

**Issue by jcnnyc** [veda-consulting/uk.co.vedaconsulting.mailchimp#145]
I am running Drupal core 7.37 / CiviCRM 4.6.2 and uk.co.vedaconsulting.mailchimp 1.7.
1. I created a list TEST2 and a group TEST2 on Mail Chimp
2. On CiviCRM, I selected a preexisting group General Interest and tried to sync with a MailChimp Interest Group.  When I tried to sync with only a MailChimp List, it works.  Below are the screen shots I get.  Any idea of how to fix this?

![screen shot 2015-05-29 at 3 43 05 pm](https://cloud.githubusercontent.com/assets/2190691/7891425/890d30e6-061a-11e5-979b-14682c73b14f.png)

![screen shot 2015-05-29 at 3 43 34 pm](https://cloud.githubusercontent.com/assets/2190691/7891428/906a847e-061a-11e5-82ab-c7984ec6bf4a.png)

**Comment by jcnnyc** [comment]
Solved.

**Comment by veda-consulting** [comment]
Hi - If you could explain what you found/did it may help others if they find themselves in the same position? It may also help us correct any issues with the installation/user guide?

**Comment by jcnnyc** [comment]
The scenario is the following:

We have parent group in CiviCRM, lets called "Advisory Board", and would like to syncs with corresponding Interest group "Advisory Board" in List "CIVICRM MAILCHIMP Contacts Sync List" on MailChimp.  This will not work. 

Resolution:

1. Create a parent group  "CIVICRM MAILCHIMP Contacts Sync List" in CiviCRM.
2. Choose the integration option, called "Sync membership of this group with membership of a Mailchimp List" and select  "CIVICRM MAILCHIMP Contacts Sync List".
3. Go to the group "Advisory Board" in CIviCRM and select group "CIVICRM MAILCHIMP Contacts Sync List" as the parent.
4. Ensure "Sync membership of with a Mailchimp interest grouping" option is set to "CIVICRM MAILCHIMP Contacts Sync List" for "Mailchimp List" and "Advisory Board::Advisory Board".  This now saves without error.
5. Sync Civi Contacts To Mailchimp.  It now syncs successfully.

In other words, in CiviCRM, you have to first create a parent group to sync with the Mailchimp list and then create a child group to sync with the Interest group in Mailchimp.  This is not ideal in our situation where we want to create one List in Mailchimp that contains all the groups from CiviCRM to sync.

Below are some of the screen shots:

![screen shot 2015-06-01 at 12 57 00 pm](https://cloud.githubusercontent.com/assets/2190691/7918247/e4fb6806-085d-11e5-9953-2f34d5fd28de.png)

![screen shot 2015-06-01 at 12 57 38 pm](https://cloud.githubusercontent.com/assets/2190691/7918252/efaa61f8-085d-11e5-889e-10b81f7c87fb.png)

**Comment by jcnnyc** [comment]
We will need to change our group hierarchy in CiviCRM in order to sync properly to the List/Group structure we want in Mailchimp.  This is a workaround and not an ideal solution.

**Comment by veda-consulting** [comment]
@jcnnyc thanks for the information. I think this should work - will take a look and see if we can find the issue.

**Comment by jcnnyc** [comment]
If it does work for you, please let me now what I'm doing wrong.  thx

**Comment by veda-consulting** [comment]
@jcnnyc after looking into this we can see that the current Mailchimp extension requires a list in order to map groups. So looks like the reconfiguration you talked about would be necessary. Going forward we may look to remove this restriction but its not a simple one as lists are used to control opt out at an organisation level. For now marking as an enhancement.

**Comment by deepak-srivastava** [comment]
We think latest v2.x has fixed this. Please re-open if the issue still exist. Thanks.
